### DISCUSSION TITLE
Contributors Wanted

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#2485] [General]
Currently we are very limited on direct contributors that are trying our best to keep this alive and moving forwards.  We really and very desperately need outside help.  While many are trying to assist, more are simply opening tickets thinking we have some funding and a more broad team working on this which is not the case.  Any contribution no matter how small will be considered.  This can be anything from spelling mistakes, hard to read documentation cleanup, fixing existing tickets, to creating new detectors.  We welcome everything.  The faster we see stuff the more likely it makes it into the next release.

**Comment by xuhuisheng** [comment]
I am trying to use spotbugs cli on our projects, and I am willing to do some contributing to the community. But I am afraid I am not familar with static analysis.

Is there any documents or paper that introduce the architecture and algorithem of spotbugs?

Thanks.

**Reply by hazendaz** [reply]
You may want to start here https://spotbugs.github.io/

**Reply by xuhuisheng** [reply]
Thanks, I add red the website of spotbugs.
And find the paper of findbugs <https://www.cs.jhu.edu/~daveho/pubs/oopsla2004.pdf>.
Keep learning.

**Comment by hazendaz** [comment]
By new contributors what I'm after is more folks that are willing to spend pull requests.  We have in fact gotten a lot of new folks being engaged in last year but we haven't yet gotten any that can become members yet that could take some of the burden we have on the 3 that are most active.  So I've pinned this.  Start slow, what we are looking for is strong developers that start with small changes then slowly increase.  Lets say 3 to 5 months if we see you enough and determined enough to get this project improved, we will invite you in with more access and somewhat expect that activity level to continue for as long as you find its possible.  Like many in this project, we move to other things in our jobs and things like this become less important a lot of times.
